packages/sitecore-jss-nextjs/src/utils.ts

```diff
@@ -77,3 +78,27 @@ export const getJssEditingSecret = (): string => {
   }
   return secret;
 };
+
+/*
+ * Pages that are statically optimized will be hydrated without their route parameters provided.
+ * After hydration, Next.js will trigger an update to your application to provide the route parameters in the query object.
+ * Details could be found on Caveats section for dynamic routes in Next.js doc
+ *
+ */
+/**
+ * @param {NextRouter} router
+ */
+export function areQueryParamsReady(router: NextRouter): boolean {
+  if (!router) {
+    return false;
+  }
+  const minQueryLength = router.query.path !== undefined ? 2 : 1;
```

Comment:
We will add a comment.
If path exists, then the minimum query length is 2, where 2 = 1 (path) + 1 (other parameter).
Otherwise, the min query length is 1 (other parameter).